homeassistant/components/unifi/config_flow.py

```diff
@@ -143,27 +156,44 @@ async def async_step_site(self, user_input=None):
                     ):
                         raise AlreadyConfigured
 
-                data = {CONF_CONTROLLER: self.config}
-
-                return self.async_create_entry(title=desc, data=data)
+                site_nice_name = self.sites[self.config[CONF_SITE_ID]]
+                return self.async_create_entry(title=site_nice_name, data=data)
 
             except AlreadyConfigured:
                 return self.async_abort(reason="already_configured")
 
         if len(self.sites) == 1:
-            self.desc = next(iter(self.sites.values()))["desc"]
-            return await self.async_step_site(user_input={})
-
-        sites = []
-        for site in self.sites.values():
-            sites.append(site["desc"])
+            return await self.async_step_site({CONF_SITE_ID: next(iter(self.sites))})
 
         return self.async_show_form(
             step_id="site",
-            data_schema=vol.Schema({vol.Required(CONF_SITE_ID): vol.In(sites)}),
+            data_schema=vol.Schema({vol.Required(CONF_SITE_ID): vol.In(self.sites)}),
             errors=errors,
         )
 
+    async def async_step_reauth(self, config_entry: dict):
+        """Trigger a reauthentication flow."""
+        self.reauth_config_entry = config_entry
+        self.reauth_config = config_entry.data[CONF_CONTROLLER]
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["title_placeholders"] = {
+            CONF_HOST: self.reauth_config[CONF_HOST],
+            CONF_SITE_ID: self.reauth_config[CONF_SITE_ID],
+        }
+
+        self.reauth_schema = {
+            vol.Required(CONF_HOST, default=self.reauth_config[CONF_HOST]): str,
+            vol.Required(CONF_USERNAME, default=self.reauth_config[CONF_USERNAME]): str,
+            vol.Required(CONF_PASSWORD, default=self.reauth_config[CONF_PASSWORD]): str,
```

Comment:
Should we empty the password field?

```suggestion
            vol.Required(CONF_PASSWORD): str,
```

Reply:
Will need to fix the tests as well